libs/shared/flagd-js-core/src/lib/feature-flag.ts

```diff
@@ -0,0 +1,202 @@
+import {FlagValue} from '@openfeature/server-sdk';
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/**
+ * Flagd flag configuration structure.
+ */
+export class FeatureFlag {
+  private readonly _state: string;
+  private readonly _defaultVariant: string;
+  private readonly _variants: Map<string, FlagValue>;
+  private readonly _targetingString: string;
+
+  constructor(flag: any) {
```

Comment:
```suggestion
  constructor(flag: unknown) {
```
`unknown` and `any` seem similar on the surface, but they are in many respects the opposite. `any` lets you do anything without type assertions, `unknown` lets you do nothing without type assertions. The latter is generally safer.

Reply:
Thank you for the explanation. I don't see any logic that goes into this class, so I think I will go ahead with the interface suggestion

Reply:
We need both the interface (to by type safe) and the class. The class is needed to parse variants and preserve targeting string

Reply:
Consider checking the new implementation - https://github.com/open-feature/js-sdk-contrib/pull/620/files#diff-6dad1c1a73438bc494e8c021c542754462aa2977828aaf2bd270927a8426ca41

Reply:
Checked it and I like it.